File: iniciante/python/1036-formula-de-bhaskara.py
import math


def delta(a, b, c):
    valor_delta = b ** 2 - 4 * a * c
    return valor_delta


def raizes(a, b, c):
    valor_de_delta = delta(a, b, c)
    if valor_de_delta < 0 or a == 0:
        return print('Impossivel calcular')
    else:
        r1 = (-b + math.sqrt(valor_de_delta))/(2*a)
        r2 = (-b - math.sqrt(valor_de_delta))/(2*a)
        return print(f'R1 = {r1:.5f}\n'
                     f'R2 = {r2:.5f}')


def entrada():
    valores = input().split(' ')
    valor_a = float(valores[0])
    valor_b = float(valores[1])
    valor_c = float(valores[2])
    return valor_a, valor_b, valor_c


n1, n2, n3 = entrada()
raizes(n1, n2, n3)

File: iniciante/c/1036-formula-de-bhaskara.c
#include <stdio.h>
#include <math.h>


float delta(float a, float b, float c){
    return pow(b, 2) - 4 * a * c; 
}

void raizes(float a, float b, float valor_de_delta){
    float r1, r2;

    r1 = (-1 * b + sqrt(valor_de_delta)) / (2 * a);
    r2 = (-1 * b - sqrt(valor_de_delta)) / (2 * a);
    
    printf("R1 = %.5f\n", r1);
    printf("R2 = %.5f\n", r2);
}

int main(){
    float valor_a, valor_b, valor_c, valor_delta;
    
    scanf("%f", &valor_a);
    scanf("%f", &valor_b);
    scanf("%f", &valor_c);

    valor_delta = delta(valor_a, valor_b, valor_c);

    if (valor_delta < 0 || valor_a == 0){
        printf("Impossivel calcular\n");
    }
    else{
        raizes(valor_a, valor_b, valor_delta);
    }
    return 0;
}
